src/app/tosBanner/tos-banner.component.ts

```diff
@@ -1,15 +1,27 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { TosBannerService } from './state/tos-banner.service';
+import { User } from '../shared/openapi';
 
 @Component({
   selector: 'app-tos-banner',
   templateUrl: './tos-banner.component.html',
   styleUrls: ['./tos-banner.component.css'],
 })
 export class TosBannerComponent {
+  @Input()
+  public bannerText: string;
+  @Input()
+  public loggedIn: boolean;
+  @Input()
+  public user: User;
```

Comment:
probably best to not pass in the user. it triggers change detection even though it's not needed until acceptTOS() is clicked on. just have the service function grab the user when needed